pallets/dapp-staking/src/migration.rs

```diff
@@ -35,116 +35,208 @@ use sp_runtime::TryRuntimeError;
+                    )
+                })?;
+
+            // Get the new values
+            let new_config = TierConfig::<T>::get();
+            let new_params = StaticTierParams::<T>::get();
+
+            // Verify that new params and new config are valid
+            assert!(new_params.is_valid());
+            assert!(new_config.is_valid());
+
+            // Verify parameters remain unchanged
             assert_eq!(
-                Pallet::<T>::on_chain_storage_version(),
-                EXPECTED_PALLET_DAPP_STAKING_VERSION,
-                "dapp-staking::migration::v9: wrong storage version"
+                old_params.slot_distribution, new_params.slot_distribution,
+                "dapp-staking-v3::migration::v10: Slot distribution has changed"
             );
-
-            assert!(
-                !ActiveProtocolState::<T>::get().maintenance,
-                "Maintenance mode must be disabled after the successful runtime upgrade."
+            assert_eq!(
+                old_params.reward_portion, new_params.reward_portion,
+                "dapp-staking-v3::migration::v10: Reward portion has changed"
+            );
+            assert_eq!(
+                old_params.tier_thresholds.len(),
+                new_params.tier_thresholds.len(),
+                "dapp-staking-v3::migration::v10: Number of tier thresholds has changed"
             );
 
-            let new_tier_params = StaticTierParams::<T>::get();
-            assert!(
-                new_tier_params.is_valid(),
-                "New tier params are invalid, re-check the values!"
+            for (_, (old_threshold, new_threshold)) in old_params
+                .tier_thresholds
+                .iter()
+                .zip(new_params.tier_thresholds.iter())
+                .enumerate()
+            {
+                match (old_threshold, new_threshold) {
+                    (
+                        TierThresholdV9::FixedPercentage {
+                            required_percentage: old_req,
+                        },
+                        TierThreshold::FixedPercentage {
+                            required_percentage: new_req,
+                        },
+                    ) => {
+                        assert_eq!(
+                            old_req, new_req,
+                            "dapp-staking-v3::migration::v10: Fixed percentage changed",
+                        );
+                    }
+                    (
+                        TierThresholdV9::DynamicPercentage {
+                            percentage: old_percentage,
+                            minimum_required_percentage: old_min,
+                        },
+                        TierThreshold::DynamicPercentage {
+                            percentage: new_percentage,
+                            minimum_required_percentage: new_min,
+                            maximum_possible_percentage: _, // We don't verify this as it's new
```

Comment:
Given the nature of this check, I'd suggest to specify only the _new_ value in the `TierThresholds: Get<[TierThreshold; 4]`. Instead of `TierThreshold`, you can receive `Option<Perbill>`.

Reply:
I've rewroked migration